### PR TITLE
Adds account lockout functionality for local accounts

### PR DESCRIPTION
**Pull request by PrahlM93**
Adds account lockout functionality for local accounts
API HTTP 400 and 404 error messages were made friendlier
Small CSS tweaks

### CODE REVIEW

**Comment by PrahlM93** [comment]
@thatarchguy, since I modified the API spec, do you remember what you used to generate the HTML from it?

**Comment by thatarchguy** [comment]
This is what i used: https://bootprint.knappi.org

File: postmaster/utils.py [hunk 1]
```diff
@@ -116,6 +137,103 @@ def add_default_configuration_settings():
         db.session.rollback()
 
 
+def is_account_unlocked(username):
```
Comment (thatarchguy) [inline postmaster/utils.py]: Would this be better on the admin model?

File: postmaster/utils.py [hunk 2]
```diff
@@ -116,6 +137,103 @@ def add_default_configuration_settings():
...
+            json_logger(
+                'error', username,
+                'The following error occurred when incrementing the failed login attempts field on "{0}": {1}'.format(
+                    username, str(e)))
+            ValidationError('A database error occurred. Please try again.', 'error')
+        finally:
+            db.session.close()
+
+
+def clear_lockout_fields_on_user(username):
```
Comment (thatarchguy) [inline postmaster/utils.py]: Same as above, would this be better on the admin model?

File: postmaster/utils.py [hunk 3]
```diff
@@ -116,6 +137,103 @@ def add_default_configuration_settings():
...
+
+        if admin.source == 'local' and admin.unlock_date and admin.unlock_date > datetime.utcnow():
+            return False
+
+        return True
+    else:
+        raise ValidationError('The user does not exist in the database.')
+
+
+def increment_failed_login(username):
```
Comment (thatarchguy) [inline postmaster/utils.py]: Same as above

**Comment by PrahlM93** [comment]
@thatarchguy thanks for the input. Can you take a look again?

**Comment by thatarchguy** [comment]
[LGTM](http://lgtm.in/i/cug77bikq)

[:+1:](http://lgtm.in/u/cug77bikq)

**Pull request merged**